CDS/scripts/dailymotion_upload.pl

```diff
@@ -188,45 +188,46 @@
 }
 
 
+sub is_imageurl_valid {
+	my ($url_to_check) = @_;
 
-my $server4 = decode_json($req->content);
+	my $url_status = 'valid';
 
-my $req;
+	if (($url_to_check eq "") || ($url_to_check eq "http://invalid.url")) {
+		$url_status = 'invalid';
+	}
+	
+	return $url_status;
+}
+    
+my $content = [
+				  title=>$videotitle,
+				  channel=>$videocat,
+				  tags=>$tags,
+				  description=>$videodescription,
+				  explicit=>$adult,
+				  access_token=>$server->{'access_token'}
+			  ];
+			  
+if ($block == 1) {
+	#$content->{'mediablocking'} = 'country/all/media/mobile';
+}
 
-if ($block == 0) {
-	my $ua = LWP::UserAgent->new;
-	$req = $ua->request(POST 'https://api.dailymotion.com/video/'.$server4->{'id'}.'?',
-		  Content_Type => 'application/x-www-form-urlencoded',
-		  Content => [
-			  title=>$videotitle,
-			  channel=>$videocat,
-			  tags=>$tags,
-			  description=>$videodescription,
-			  explicit=>$adult,
-			  thumbnail_url=>$imageurl,
-			  #taken_time=>$videotaken,
-			  access_token=>$server->{'access_token'}
-		  ]
-	);
+if (is_imageurl_valid($imageurl) eq 'valid') {
```

Comment:
if you return a boolean value from is_imageurl_valid, this can be written if(is_imageurl_valid($imageurl)){ which is nicer

Reply:
Done.